### DISCUSSION TITLE
Jinja templating in decorated task - loading external files

### DISCUSSION THREAD

**Discussion by curious-bug** [apache/airflow#39380] [Q&A] [answered]
Version: 2.7.3

I have a python function with the task.short_circuit decorator. This task uses several Boolean parameters and an SQL query to check if downstream tasks should be skipped
```python
@task.short_circuit(task_id='check_run_conditions')
def check_task(bool_1, bool_2, bool_3, sql_query = "path/to/query.sql"):
   ...
```

In normal airflow operators I have been able to use the `template_ext` parameter to have jinja automatically load my sql files in for the operator to use.

```python
check_db = CustomSQLTaskOperator(
    task_id="check_task",
    ...
    sql="/path/to/query.sql"
)
```
However, I have not found a way to make a decorated task do the same.

__Is there a way to have jinja load content from a file in a python decorated task?__

**Comment by curious-bug** [comment]
Forgot to mention- while I can just include the SQL query as a string in python itself, it is preferred to leave it out due to its length and general project desires.

**Reply by Taragolis** [reply, marked as answer]
https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html#templating

**Reply by curious-bug** [reply]
Thank you, that worked
Slightly annoying that the task decorator and BaseOperator use different argument names but whatever
